csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.Nuget.cs

```diff
@@ -14,6 +16,13 @@ private void RestoreNugetPackages(List<FileInfo> allNonBinaryFiles, IEnumerable<
         {
             try
             {
+                var checkNugetFeedResponsiveness = EnvironmentVariables.GetBoolean(EnvironmentVariableNames.CheckNugetFeedResponsiveness);
+                if (checkNugetFeedResponsiveness && !CheckFeeds(allNonBinaryFiles))
+                {
+                    DownloadMissingPackages(allNonBinaryFiles, dllPaths, withNugetConfig: false);
```

Comment:
If we just directly to here, the all packages will implicitly attempt to be installed by creating a new dotnet project and adding the package as reference before attempting a restore. This happens in temporary directory, but could it be the case that a user or machine specific `nuget.config` file could be applied that also suffers from an unresponsive feed?

Reply:
This is a possibility. `CheckFeeds` is only validating feeds that are explicitly in nuget.config files in the analyzed directory. We're not checking user or machine specific ones.

Reply:
That is probably an acceptable risk. It can be improved in case we run into this issue.